### PR TITLE
geojson: use more exact type for Position

### PR DESCRIPTION
**Pull request by EvanHahn**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test geojson`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.1>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/geojson/geojson-tests.ts [hunk 1]
```diff
@@ -11,8 +11,17 @@ import {
     MultiPolygon,
     Point,
     Polygon,
+    Position,
 } from "geojson";
 
+const position1: Position = [1, 2];
+const position2: Position = [1, 2, 3];
+const legacyPosition: Position = [1, 2, 3, 4];
```
Comment (EvanHahn) [inline types/geojson/geojson-tests.ts:19]: As an alternative, I could prohibit legacy positions and change the type to `[number, number] | [number, number, number]`.

File: types/shpjs/shpjs-tests.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ shp.getShapefile(new Int32Array(50)).then((geojson) => {});
 const combinedGeojson = shp.combine([
     [{
         type: "Point",
-        coordinates: [],
+        coordinates: [10, 20],
```
Comment (EvanHahn) [inline types/shpjs/shpjs-tests.ts:22]: This type was incorrect as far as I could tell.

**Comment by danvk** [comment]
This exact change was proposed, discussed and rejected two months ago, see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70029#issuecomment-2231830232

**Comment by EvanHahn** [comment]
Thanks for sharing, I didn't see that.

I'll close this.

**Pull request closed without merging**